repository_service_tuf/cli/admin/ceremony.py

```diff
@@ -638,6 +638,15 @@ def _run_ceremony_steps(save: bool) -> Dict[str, Any]:
             f" {pending_signatures} pending signature(s) left.",
         )
 
+    required_num_of_keys = (
+        setup.threshold[Roles.ROOT] - setup.number_of_keys[Roles.ROOT]
```

Comment:
It's important to be clear with the terminology here.
For DAS we expect that the number of **`LOADED`** keys will be less than the threshold, but the number of keys that we have **`SETUP`** for root should be equal to or more than the threshold.
The number of keys we have **`SETUP`** is equal to the number of  **`LOADED`** keys + the number of keys for which we have provided public key information.

Here we want to print the difference between how many keys we have **`SETUP`** minus the number of expected root keys in total.
I think you shouldn't use here `setup.threshold[Roles.ROOT]`, but instead `len(metadata[Roles.ROOT.value].signed.keys)`.
The `metadata[Roles.ROOT.value].signed.keys` contains all **`LOADED`**  keys + those for which we have given public key value.

Am I correct with my understanding here @kairoaraujo?

Reply:
Another mistake I see: it should be `setup.number_of_keys[Roles.ROOT] - len(metadata[Roles.ROOT.value].signed.keys)` otherwise the result will be 0 or negative.

Reply:
> Am I correct with my understanding here @kairoaraujo?

Yes, It is correct. I'm adding some simplification.

> It's important to be clear with the terminology here. For DAS we expect that the number of **`LOADED`** keys will be less than the threshold, but the number of keys that we have **`SETUP`** for root should be equal to or more than the threshold.

IMO, we don't need to care if it is DAS or Signed process, but check what you suggested below.

> The number of keys we have **`SETUP`** is equal to the number of **`LOADED`** keys + the number of keys for which we have provided public key information.
> 
> Here we want to print the difference between how many keys we have **`SETUP`** minus the number of expected root keys in total. I think you shouldn't use here `setup.threshold[Roles.ROOT]`, but instead `len(metadata[Roles.ROOT.value].signed.keys)`. The `metadata[Roles.ROOT.value].signed.keys` contains all **`LOADED`** keys + those for which we have given public key value.
>

Reply:
This use-case, i.e. `(A)setup.number_of_keys[Roles.ROOT] - (B)len(metadata[Roles.ROOT.value].signed.keys)` will never be >0 as `signed.keys` includes all the keys, not only the ones used to sign root. There will always be at least one online key. 
* If (A) == 1 (instead of the default 2), (A) - (B) will be <0, as (B) will be at least 1 root key (ceremony would fail prior otherwise) + at least 1 online key. 
* If (A) == 2, (B) would be again >= 2 (at least 1 root key + at least 1 online key)
* If (A) > 2, ceremony would fail prior without having configured a metadata, that has >2 root keys - one cannot give a setup number of root keys without passing them as ceremony input afterwards, so this will result to a metadata that has (A) number of keys in and we go to the previous 2 use-cases - we'd just increase the numbers.

If we use only the loaded keys used to sign root for (B) instead of `metadata[Roles.ROOT.value].signed.keys`, (A)-(B) will always be 0, as during ceremony one would pass as many root keys as (A) to generate the root metadata. Otherwise, ceremony would fail prior this check.

This makes a so defined `required_num_of_keys` useless. 
This is why it's now changed to compare threshold to the loaded keys used to sign root, instead of `setup.number_of_keys[Roles.ROOT]`, only for functional clarity, even though they would be practically equal.

Reply:
In this context `required_num_of_keys` will always be equal to `pending_signatures` out of the DAS context.

Moved this check prior the check for `pending_signatures`, as the latter will always be >=`required_num_of_keys` in the DAS context.

Reply:
We had a live chat with @MVrachev and both agreed on this approach.

@kairoaraujo Could you please share your point on that? Thanks

Reply:
> If (A) > 2, ceremony would fail prior for not having configured a metadata, that has >2 root keys - one cannot give a setup number of root keys without passing them as ceremony input afterwards, so this will result to a metadata that has (A) number of keys in and we go to the previous 2 use-cases - we'd just increase the numbers.

About this case: we make it mandatory to input the same amount of keys as you have entered as `number_of_keys`.